spec/tools/test_inventory_generator

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env ruby
+
+require "json"
+
+require "topological_inventory/ingress_api/client"
+include TopologicalInventory::IngressApi::Client
+
+test_inventory_dir = File.expand_path(File.join("..", "persister", "test_inventory"), __dir__)
+
+container_project = ContainerProject.new(
+  :source_ref        => "c56c6854-baaf-11e8-ba7e-d094660d31fb",
+  :name              => "default",
+  :display_name      => "Default",
+  :resource_version  => "42001363",
+)
+
+container_node = ContainerNode.new(
+  :source_ref        => "b1bb5472-675d-4939-89b4-00a016fc0556",
+  :name              => "node",
+  :cpus              => 4,
+  :memory            => 17179869184,
+  :resource_version  => "25007341",
+)
+
+simple_inventory = Inventory.new(
+  :name        => "OCP",
+  :source      => "9a874712-9a55-49ab-a46a-c823acc35503",
+  :schema      => Schema.new(:name => "Default"),
+  :collections => [
+    InventoryCollection.new(
+      :name => :container_projects,
+      :data => [container_project]
+    ),
+  ]
+)
+
+File.write(File.join(test_inventory_dir, "simple_inventory.json"), JSON.dump(simple_inventory.to_hash))
+
+relationships_inventory = Inventory.new(
+  :name        => "OCP",
+  :source      => "9a874712-9a55-49ab-a46a-c823acc35503",
+  :schema      => Schema.new(:name => "Default"),
+  :collections => [
+    InventoryCollection.new(
+      :name => :container_projects,
+      :data => [container_project],
+    ),
+    InventoryCollection.new(
+      :name => :container_nodes,
+      :data => [container_node],
+    ),
+    InventoryCollection.new(
+      :name => :container_groups,
+      :data => [
+        ContainerGroup.new(
+          :source_ref        => "a40e2927-77b8-487e-92bb-63f32989b015",
+          :name              => "pod",
+          :resource_version  => "1432020",
+          :container_project => InventoryObjectLazy.new(
+            :inventory_collection_name => :container_projects,
+            :reference                 => {:name => container_project.name},
+            :ref                       => :by_name,
+          ),
+          :container_node    => InventoryObjectLazy.new(
+            :inventory_collection_name => :container_nodes,
+            :reference                 => {:name => container_node.name},
+            :ref                       => :by_name,
+          )
+        )
+      ]
+    ),
+  ]
+)
+
+File.write(File.join(test_inventory_dir, "relationships_inventory.json"), JSON.dump(relationships_inventory.to_hash))
```

Comment:
Can all of this live in a module with something like `TestInventoryGenerator.run if __FILE__ == $PROGRAM_NAME` at the end?  Then if the file is required, it won't unintentionally do anything.

Reply:
Yeah sure thing